asserts/tool/sign.go

```diff
@@ -0,0 +1,91 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+// Package tool offers tooling to sign assertions.
+package tool
+
+import (
+	"fmt"
+
+	"gopkg.in/yaml.v2"
+
+	"github.com/snapcore/snapd/asserts"
+)
+
+// SignRequest specifies the complete input for signing an assertion.
+type SignRequest struct {
+	// KeyID specifies the key id of the key to use
+	KeyID string
+
+	// Statement is used as input to construct the assertion
+	// it's a mapping encoded as YAML
+	// of the header fields of the assertion
+	// plus an optional pseudo-header "body" to specify
+	// the body of the assertion
+	Statement []byte
+}
+
+// Sign produces the text of a signed assertion as specified by req.
+func Sign(req *SignRequest, keypairMgr asserts.KeypairManager) ([]byte, error) {
```

Comment:
Will `SignRequest` grow more struct members? It seems like for just two we might as well have `sign(key, statement, keypairmgr)`. Of course if it will grow or if there are other good reasons thats totally fine, just wondering a bit :)